### DISCUSSION TITLE
Changing -memberlist.advertise-addr in Docker

### DISCUSSION THREAD

**Discussion by ntex777** [grafana/mimir#9108] [Help and support] [answered]
Based on this discussion [Docker DNS Service Discovery "no such host"](https://github.com/grafana/mimir/discussions/4517) while running Mimir on 3 node cluster, how to achieve the same result on Distroless ?
Since we can't sh, since release r294, distroless became main:
`OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"sh\": executable file not found in $PATH": unknown.`

I have an entrypoint script that does few before starting Mimir.

Before you suggest the mimir-alpine release, I'm aware of it's existence, but I'm also aware they will deprecate this release in near future, so I'm just trying to avoid more issues that needs to be tackled in future.

I know in the future, I could just build my own image, based on this git, but I'm trying to avoid this as well.

Thanks for your suggestions.

**Comment by pstibrany** [comment, marked as answer]
We have decided to support only deployments on Kubernetes. For other purposes, you will need to build custom image (or use `mimir-alpine` for now, but as you point out, that's temporary solution).

For more details see: https://github.com/grafana/mimir/issues/9034#issuecomment-2304042358

**Reply by ntex777** [reply]
Thanks for your quick reply.

I would love also to move to Kubernetes entirely and drop Swarm! Specially in 2024, but I still can't make that move, unfortunately.

So, this is main [Dockerfile](https://github.com/grafana/mimir/blob/26c1c9ac4561a33a7845f644a06b1b91795df320/cmd/mimir/Dockerfile), to building from scratch ? 
And then I can perhaps just add the sh from Busybox, should be enough.

Thanks

**Reply by pstibrany** [reply]
Yes, that is the main Dockerfile. You may also simply change the base image to say `alpine:3.19.1` (by passing `BASEIMG` argument) instead of adding the package.

**Comment by ntex777** [comment]
Thank you. If I use Alpine will use 3.20 instead, since has 0 CVE's or just Busybox 1.36 (uclibc).
In the end, we don't need much, just sh, ls, cat, ifconfig, cut, grep, awk, that's it so the entrypoint can get the IP of the running container and pass to the cluster as Advertised Address, as you know this only happens for Docker.
